taskpools/chase_lev_deques.nim

```diff
@@ -148,6 +150,10 @@ proc push*[T](deque: var ChaseLevDeque[T], item: T) =
     # Full queue
     deque.grow(a, t, b)
 
+  if not deque.garbage.isNil and b == t:
+    # Empty queue, no thieves can have a pointer to an old retired buffer
```

Comment:
Assumption seems correct to me, if `steal` copies `rawBuffer`.

If `steal` _were_ `lent` instead, the `c_free(node)` could mess with it, right?